File: js/RockPaperScissor.js
// function rps(p1, p2){
    
//     return p1 === p2 ? "Draw" : (p1 === "rock" && p2 === "scissors" || p1 === "scissors" && p2 === "paper" || p1 === "paper" && p2 === "rock") ? "Player 1 won!" : "Player 2 won!"
// }


function rps(p1, p2) {
    const opcoes = ["rock", "paper", "scissors"];

    if (p1 == p2) {
        return "Draw";
    }

    if (opcoes.indexOf(p1) !== -1 && opcoes.indexOf(p2)!== -1) {
        return (opcoes.indexOf(p1) + 1) % 3 === opcoes.indexOf(p2) ? "Player 2 won!" : "Player 1 won!";
    } else {
        return "Escolha inválida! Por favor, escolha entre rock, paper ou scissors.";
    }
}
console.log(rps("rock", "scissors"))
console.log(rps("scissors", "scissors"))
console.log(rps("paper", "scissors"))
console.log(rps("paper", "rock"))

File: python/RockPaperScissor.py
def rps(p1, p2):
    rules = ['rock', 'paper', 'scissors']
    if p1 == p2:
        return 'Draw!'
    elif rules[(rules.index(p1) + 1) % 3] == p2:
        return 'Player 2 won!'
    else:
        return 'Player 1 won!'


# def rps(p1, p2):
#     rules={'rock':'scissors', 'paper':'rock', 'scissors':'paper'}
#     if p1==p2:
#         return 'Draw!'
#     elif p1==rules[p2]:
#         return 'Player 2 won!'
#     else:
#         return 'Player 1 won!'


print(rps("rock", "scissors"))